### PR TITLE
[ci] Upstream benchmark script changes from https://github.com/paritytech/substrate/pull/11493

### PR DESCRIPTION
**Pull request by coderobe**
See https://github.com/paritytech/substrate/pull/11493

### CODE REVIEW

File: scripts/ci/run_benches_for_runtime.sh [hunk 1]
```diff
@@ -1,54 +1,72 @@
...
-# This true makes sure that $? is 0 instead of
-# carrying over a failure which would otherwise cause
-# the whole CI job to abort.
-true
+# Check if the error file exists.
+if [ -f "$ERR_FILE" ]; then
+  echo "[-] Some benchmarks failed. See: $ERR_FILE"
+else
+  echo "[+] All benchmarks passed."
+fi
```
Comment (chevdor) [inline scripts/ci/run_benches_for_runtime.sh:72]: eof

Comment (chevdor) [inline scripts/ci/run_benches_for_runtime.sh:72]:
```suggestion
fi

```

**Review by chevdor — APPROVED** [review]
LGTM besides the missing eof

File: scripts/ci/run_benches_for_runtime.sh [hunk 2]
```diff
@@ -1,54 +1,72 @@
...
+    ./target/production/polkadot benchmark pallet \
     --chain="${runtime}-dev" \
     --steps=50 \
     --repeat=20 \
-    --pallet="$pallet" \
+    --pallet="$PALLET" \
     --extrinsic="*" \
     --execution=wasm \
     --wasm-execution=compiled \
     --heap-pages=4096 \
```
Comment (ggwpez) [inline scripts/ci/run_benches_for_runtime.sh:41]: I think we can remove this now.

**Review by ggwpez — APPROVED** [review]

File: scripts/ci/run_benches_for_runtime.sh (hunk 2 above)
Comment (coderobe) [reply scripts/ci/run_benches_for_runtime.sh:41]:
```suggestion
```

**Review by sergejparity — APPROVED** [review]

**Review by EgorPopelyaev — APPROVED** [review]

**Comment by coderobe** [comment]
bot merge

**Comment by coderobe** [comment]
bot merge

**Pull request merged**